tests/kernel/device/src/mmio_multireg.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ * Copyright (c) 2022 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <zephyr/device.h>
+
+#define DT_DRV_COMPAT	fakedriver_multireg
+
+/*
+ * Driver with multiple MMIO regions to manage defined into DT
+ */
+
+struct foo_multireg_dev_data {
+	int baz;
+
+	DEVICE_MMIO_NAMED_RAM(chip);
+	DEVICE_MMIO_NAMED_RAM(dale);
+};
+
+struct foo_multireg_dev_data foo_multireg_data;
+
+struct foo_multireg_config_info {
+	DEVICE_MMIO_NAMED_ROM(chip);
```

Comment:
Just a quick question, do we have a `_FOR_EACH ` macro? Cause I think it will be tedious if we have tons of regs, LOL.
But definitely, it is not the work this PR should do.

Reply:
yes we have :)